hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCacheTest.java

```diff
@@ -161,4 +161,24 @@ public void testNearCacheIdleRecordsExpired_withObjectInMemoryFormat() {
     public void testNearCacheIdleRecordsExpired_withBinaryInMemoryFormat() {
         testNearCacheExpiration_withMaxIdle(InMemoryFormat.BINARY);
     }
+
+    @Test
+    public void testNearCacheMemoryCostCalculation_withObjectInMemoryFormat() {
```

Comment:
This test class could be a good candidate for deduplication using parametrization (perhaps in another PR?).